lxd/operations.go

```diff
@@ -54,7 +54,20 @@ var operationWebsocket = APIEndpoint{
 // that when reached will shut down the instances forcefully.
 // It also watches the cancel channel, and will return if it receives data.
 func waitForOperations(s *state.State, chCancel chan struct{}) {
-	timeout := time.After(5 * time.Minute)
+	var timeout <-chan time.Time
+	err := s.Cluster.Transaction(func(tx *db.ClusterTx) error {
+		config, err := cluster.ConfigLoad(tx)
+		if err != nil {
+			return err
+		}
+		timeout = time.After(config.ShutdownTimeout())
+		return nil
+	})
+	if err != nil {
+		// If something goes really bad just set timeout to 5 minutes
```

Comment:
@presztak please could you log an error here as well as setting the default timeout so we can see if something goes wrong. Thanks. e.g `logger.Error("Failed getting shutdown timeout", log.Ctx{"err": err})`

Reply:
@tomponline logging implemented.